### PR TITLE
Testing

### PR DESCRIPTION
**Pull request by sooryan**
Addresses #4

### CODE REVIEW

File: src/blake2b.c [hunk 1]
```diff
@@ -189,7 +189,7 @@ blake2b_init(blake2b_state* state, size_t outlen, const void* key, size_t keylen
   if (keylen > 0) {
     uint8_t block[BLAKE2B_BLOCKBYTES] = {0};
     memcpy(block, key, keylen);
-    blake2b_update(state, block, BLAKE2B_BLOCKBYTES);
+    blake2b_update(&state, block, BLAKE2B_BLOCKBYTES);
```
Comment (Sachin-A) [inline src/blake2b.c]: Why is a pointer to the state pointer being passed here? Incompatible pointer type

File: src/test.c [hunk 2]
```diff
@@ -14,38 +13,41 @@ print_hex(const uint8_t* hash, char* string, int len)
...
+  for (i = 0; i < BLAKE2_KAT_LENGTH; ++i) {
     blake2b(hash, BLAKE2B_OUTBYTES, buf, i, key, BLAKE2B_KEYBYTES);
+    if (memcmp(hash, blake2b_kat[i], BLAKE2B_OUTBYTES)) {
+      printf("%d\n", (int)i);
+      printf("Failed\n\n");
+      print_hex(buf, "buffer", i);
+      print_hex(key, "key", 0);
+      print_hex(hash, "output", BLAKE2B_OUTBYTES);
+      print_hex(blake2b_kat[i], "expected", BLAKE2B_OUTBYTES);
+      return -1;
```
Comment (Sachin-A) [inline src/test.c]: Better to keep a log of all the failed cases than exit at the first failure.

File: src/test.c [hunk 3]
```diff
@@ -5,35 +5,47 @@
...
+    buf[i] = (uint8_t)i;
+
+  for (i = offset; i < BUF_LENGTH; ++i)
+    buf[i] = (uint8_t)0;
 
-  for (i = 0; i <= buflen; ++i) {
-    blake2b(hash, BLAKE2B_OUTBYTES, buf, i, key, keylen);
+  /* Test simple API */
+  for (i = 0; i < offset; ++i) {
+    blake2b(hash, BLAKE2B_OUTBYTES, buf, i, key, BLAKE2B_KEYBYTES);
```
Comment (Sachin-A) [inline src/test.c]: Why is key size passed as BLAKE2B_KEYBYTES when no key is used?

File: src/test.c [hunk 4]
```diff
@@ -1,39 +1,53 @@
...
+    key[i] = (uint8_t)i;
+
+  for (i = 0; i < BLAKE2_KAT_LENGTH; ++i)
+    buf[i] = (uint8_t)0;
+
+  for (i = 0; i < BLAKE2_KAT_LENGTH; ++i)
+    buf[i] = (uint8_t)i;
+
+  for (i = 0; i < BLAKE2_KAT_LENGTH; ++i) {
+    blake2b(hash, BLAKE2B_OUTBYTES, buf, i, key, BLAKE2B_KEYBYTES);
```
Comment (Sachin-A) [inline src/test.c]: Why is key size passed as BLAKE2B_KEYBYTES when no key is used?

File: src/blake2b.c [hunk 5]
```diff
@@ -201,22 +201,30 @@ blake2b_init(blake2b_state* state, size_t outlen, const void* key, size_t keylen
...
+blake2b_update(blake2b_state* state, const unsigned char* input_buffer,
+               size_t inlen)
 {
-  const unsigned char* in;
-
-  in = input_buffer;
-
-  while (inlen > BLAKE2B_BLOCKBYTES) {
+  unsigned char* in = input_buffer;
+  blake2b_state* state
```
Comment (Sachin-A) [inline src/blake2b.c]: Missing semicolon

Comment (sooryan) [reply src/blake2b.c]: I'm pretty sure I checked. Not sure how I missed this. Will change.

File: src/test.c (hunk 2 above)
Comment (sooryan) [reply src/test.c]: Sure, removing the return and piping the output to a file should do I guess

File: src/blake2b.c (hunk 1 above)
Comment (sooryan) [reply src/blake2b.c]: Oddly enough, when not done this way, I get wrong results. 
Not exactly sure why though

File: src/test.c (hunk 4 above)
Comment (sooryan) [reply src/test.c]: It shouldn't really matter seeing as how there's no key. I'll make the change anyway.

File: src/blake2b.c (hunk 1 above)
Comment (SooryaN) [inline src/blake2b.c]: Ah, I should be passing the key here. My bad.

**Comment by Sachin-A** [comment]
Outdated.
Check PR #8

**Pull request closed without merging**